src/kili/adapters/kili_api_gateway/asset/formatters.py

```diff
@@ -0,0 +1,21 @@
+"""Formatters for assets retrieved from Kili API."""
+
+import json
+from typing import Dict
+
+from kili.domain.types import ListOrTuple
+
+ASSET_JSON_FIELDS = ["jsonMetadata", "labels.jsonResponse", "latestLabel.jsonResponse"]
+
+
+def load_asset_json_fields(asset: Dict, fields: ListOrTuple[str]) -> Dict:
+    """Load json fields of an asset."""
+    if "jsonMetadata" in asset:
+        asset["jsonMetadata"] = json.loads(asset["jsonMetadata"])
+    if "labels.jsonResponse" in fields:
+        asset_labels = asset.get("labels", [])
+        for label in asset_labels:
+            label["jsonResponse"] = json.loads(label["jsonResponse"])
+    if "latestLabel.jsonResponse" in fields and asset.get("latestLabel") is not None:
+        asset["latestLabel"]["jsonResponse"] = json.loads(asset["latestLabel"]["jsonResponse"])
+    return asset
```

Comment:
I would be super careful here, some fields might actually be None when they are not mandatory in the graphql schema, even though the default value is different.

For example, jsonMetadata is not mandatory, so it might be None, but in theory, we should fetch "{}" instead.

But I trust the schema better than the backend behavior, so I would still use some `.get()` for the dictionaries, and check that the value is a string before putting it in the json.loads

Reply:
Yeah, safer indeed